### DISCUSSION TITLE
Shoreline Filtering

### DISCUSSION THREAD

**Discussion by ddbuscombe** [SatelliteShorelines/CoastSeg#132] [General]
A space to discuss implementation of shoreline filtering

From @2320sharon:

additional filters we could add and here are some ideas I have based on what you all have said + my own research/inspiration.

  *  I notice a common problem are little islands of shorelines caused by misclassification we could have a switch that the user could turn on and off to get rid of islands.
*    Instead of relying only on the shoreline length being longer than the minimum beach length parameter set by the user we could make sure the extracted shorelines are not dramatically longer or shorter than the reference shoreline.
  *  We could use kmeans clustering to group shorelines together by the the curvature of the shoreline, hausdorff_distance (or another metric), and length to find which shorelines are most similar to each other and the reference shoreline and drop the rest. This was suggested by both Dan and Evan :brain: 

Adding that single filter made a massive difference on my truc vert site along with using the 4 class model. I also increased the length of the minimum shoreline as well as had the the percent bad pixels allowed at 14%.
Extracting Shorelines seems to consistently fail on L7 imagery so these settings filter them out. 

![image](https://github.com/Doodleverse/CoastSeg/assets/3596509/900ce550-b1d1-4bea-bf27-db5a0b9e5b00)

One thing we need to emphasize is the importance of using the minimum shoreline length filter and the percent no data filter since these two make a big difference in the quality of the extracted shorelines

**Comment by ddbuscombe** [comment]
I tried to do a k means clustering but couldn't get it to work. The approach I used was to interpolate each Shoreline to a common axis, then take the difference of that shoreline with respect to the reference shoreline. I then tried to do kmeans with and without vector whitening, and the clusters it gave me weren't useful. Maybe a different approach?

**Comment by ddbuscombe** [comment]
https://github.com/Doodleverse/CoastSeg/issues/130

**Comment by 2320sharon** [comment]
This example image shows what a difference filters make by adding the no data filter and a  higher threshold for the shoreline length you can see that a bad shoreline, which is shown in red, eliminated. Granted this image still has some bad shoreline, but a simple filter to drop shoreline points outside a certain range from the reference shoreline easily fixes this issue.
![image](https://github.com/Doodleverse/CoastSeg/assets/61564689/c8e0f320-aef9-4787-8ba1-c7b0fa2735dc)
